File: src/LletresRepetidesJavaScript/Fase5.js
/**
 * Fase5 - Ejercico de LletresRepetides en JavaScript
 * @author Rubén Rodríguez
 * IT Academy
 * Barcelona Activa
 *
 */
let listNombre5 = ['R','u','b','é','n']
let listCognom5 = ['R','o','d','r','í','g','u','e','z']
let nomComplert5 = []
let impresFs5 = false

var respostaFase5 = function(){

	//Comprobamos que no se ha impreso por pantalla. En caso afirmativo no hace nada.

	if (impresFs5){}
	else {
		impresFs5 = true
		let selectElement = document.getElementById('fase5Sortida')

		nomComplert5 = listNombre5.concat('\xa0').concat(listCognom5)

		//Imprimimos en el html la lista.

		nomComplert5.forEach(function(letra) {
			selectElement.innerHTML += letra+" "})
			
	}
}

//Nos aseguramos de cargar todo el JS al cargar el document html.

if (document.addEventListener){
    window.addEventListener('load',(function() {
    	document.getElementById('5').onclick = respostaFase5
    }),false)
} else {
    window.attachEvent('onload',(function() {
    	document.getElementById('5').onclick = respostaFase5
    }))  
}

File: src/lletreRepetidesJava/Fase5.java
/**
 * Fase5 - Ejercico de LletresRepetides
 * @author Rubén Rodríguez
 * IT Academy
 * Barcelona Activa
 *
 */

package lletreRepetidesJava;

import java.util.ArrayList;
import java.util.List;

public class Fase5 {
	
	private List<Character> listNombre = new ArrayList<Character>();
	private List<Character> listCognom = new ArrayList<Character>();
	private List<Character> nomComplert = new ArrayList<Character>();
	
	protected Fase5() {
		listNombre.add('R');
		listNombre.add('u');
		listNombre.add('b');
		listNombre.add('e');
		listNombre.add('n');
		listCognom.add('R');
		listCognom.add('o');
		listCognom.add('d');
		listCognom.add('r');
		listCognom.add('i');
		listCognom.add('g');
		listCognom.add('u');
		listCognom.add('e');
		listCognom.add('z');
	}
	
	
	protected void fase5Print () {
		
		//Concatenamos las listas en una sola
		
		nomComplert.addAll(listNombre);
		nomComplert.add(' ');
		nomComplert.addAll(listCognom);
		
		nomComplert.forEach(letra->{
			System.out.print(letra);
		});
		
	}
	
}